src/odemis/acq/stream/_projection.py

```diff
@@ -179,6 +162,112 @@ def needImageUpdate(self):
         self._projectedTilesInvalid = True
         self._shouldUpdateImage()
 
+    def onTint(self, value):
+        if isinstance(self.raw, list):
+            if len(self.stream.raw) > 0:
+                raw = self.stream.raw[0]
+            else:
+                raw = None
+        elif isinstance(self.stream.raw, tuple):
+            raw = self.stream._das
+        else:
+            raise AttributeError(".raw must be a list of DA/DAS or a tuple of tuple of DA")
+
+        if raw is not None:
+            # If the image is pyramidal, the exported image is based on tiles from .raw.
+            # And the metadata from raw will be used to generate the metadata of the merged
+            # image from the tiles. So, in the end, the exported image metadata will be based
+            # on the raw metadata
+            raw.metadata[model.MD_USER_TINT] = value
+
+        self._shouldUpdateImage()
+
+    def _project2RGB(self, data, tint=(255, 255, 255)):
+        """
+        Project a 2D DataArray into a RGB representation
+        data (DataArray): 2D DataArray
+        tint ((int, int, int)): colouration of the image, in RGB.
+        return (DataArray): 3D DataArray
+        """
+        # TODO replace by local irange
+        irange = self.stream._getDisplayIRange()
+        rgbim = img.DataArray2RGB(data, irange, tint)
+        rgbim.flags.writeable = False
+        # Commented to prevent log flooding
+        # if model.MD_ACQ_DATE in data.metadata:
+        #     logging.debug("Computed RGB projection %g s after acquisition",
+        #                    time.time() - data.metadata[model.MD_ACQ_DATE])
+        md = self._find_metadata(data.metadata)
+        md[model.MD_DIMS] = "YXC"  # RGB format
+        return model.DataArray(rgbim, md)
+
+    def _updateImage(self):
+        """ Recomputes the image with all the raw data available
+        """
+        # logging.debug("Updating image")
+        if not self.stream.raw and isinstance(self.stream.raw, list):
+            return
+
+        try:
+            # if .raw is a list of DataArray, .image is a complete image
+            if isinstance(self.stream.raw, list):
+                raw = self.stream.raw[0]
+
+            raw = img.ensure2DImage(raw)
+            self.image.value = self._project2RGB(raw, self.stream.tint.value)
+
+        except Exception:
+            logging.exception("Updating %s %s image", self.__class__.__name__, self.name.value)
+
+
+class RGBSpatialProjection(RGBProjection):
+
+    def __new__(cls, stream):
+
+        if isinstance(stream, StaticSpectrumStream):
+            return super(RGBSpatialProjection, cls).__new__(RGBSpatialSpectrumProjection, stream)
```

Comment:
Explain a little bit why we do this trick.
=> Special code to handle the spectrum, which we don't want to mix with the rest of the code to keep things simple.